docs/rfcs/20210223_external_connectivity_in_k8s.md

```diff
@@ -0,0 +1,203 @@
+brokers addresses should be as quick as possible.
+
+## What use cases does it support?
+
+Exposing each Redpanda node to the internet.
+
+## What is the expected outcome?
+
+The Redpanda operator will make sure that from the network standpoint
+brokers will be reachable. Users need to managed security of the
+exposed Kubernetes worker nodes e.g. `aws security groups`.
+
+# Guide-level explanation
+## How do we teach this?
+
+The Kafka API requires that each broker will be reachable from
+the client side. To expose Redpanda nodes, but not add additional
+latency to the communication each Redpanda process will have assigned
+unique host port. In case of cloud providers the host VM need to be
+publicly reachable and have opened port. The operator would be
+responsible to orchestrate Kubernetes to achieve that.
+
+## Dictionary
+
+- Kubernetes Service type NodePort with empty selector that gives
+  unused port that is available across all k8s worker nodes
+
+# Reference-level explanation
+
+( This is the technical portion of the RFC )
+
+## Interaction with other features
+
+- The kafka multi listener needs to be set up in each Redpanda
+  node, so that internal connectivity can stay as is. For external
+  connectivity the dedicated port would be opened.
+- The statefulset must bootstrap Redpanda cluster with or without
+  external connectivity.
+- The user intention defined in cluster.redpanda.vectorized.io custom
+  resource should be reconciled into set of ready to use address with
+  ports that can be reach from the internet. The status field will
+  be the place to look for the addresses.
+
+## Telemetry & Observability
+
+- metrics that can be anchor in Redpanda or in VM network stack
+    - Number of active connection
+    - Number of drpped connection
+    - Number of bytes processed
+- Tail latency observed by the clients
+- Rate of changes in cloud provider DNS service
+- Number of DNS registry errors in `external-dns`
+- Additional probe that will live close to operator, but will try
+  reach the Redpanda Nodes by their external DNS address
+
+## Corner cases dissected by example.
+
+- List all corner cases:
+  And a detailed explanation of the corner cases
+
+## Detailed design - What needs to change to get there
+
+The host port need to be added in kafka port in the Statefulset
+definition. Service type node port need to be added.
+
+## Detailed design - How it works
+
+Each Kubernetes node needs to be exposed to the internet in order
+to expose Redpanda node. Operator responsibility is creation of
+Redpanda configuration that has correctly registered advertised
+Kafka API and assign correct unique and unused node port. With the
+help of Kubernetes Service type node port the unused port can be
+injected into container host port. This service doesn't have selector,
+so that no rule should be created in `iptable` or any other CNI
+implementation.
+
+## Drawbacks
+
+If the Redpanda POD will need to be rescheduled to different
```

Comment:
how is this related to this RFC? 🤔

Reply:
Agree, my thoughts gone in wrong direction. Removed this paragraph